components/cookie-consent/cookie-consent.tsx

```diff
@@ -0,0 +1,64 @@
+import { PageLinks } from '../../shared/config'
+import { getCookie, setCookie } from 'nextjs-cookie'
+import { useEffect, useState } from 'react'
+import { COOKIE_CONSENT_KEY, Consent } from './cookie-consent.definitions'
+import { updateConsent } from '../../shared/gtag'
+
+export const CookieConsent = (): JSX.Element | null => {
+  const [cookieConsent, setCookieConsent] = useState<boolean>(true)
+
+  useEffect(() => {
+    const cookieConsent = getCookie(COOKIE_CONSENT_KEY)
+    if (cookieConsent) {
+      setCookieConsent(true)
+    } else {
+      setCookieConsent(false)
+    }
+  }, [])
+
+  if (cookieConsent) {
+    return null
+  }
+
+  const handleClick = (
```

Comment:
What about using `useCallback` ?

Reply:
Uhuh, nice to see you around. Let me add that indeed :)